okcupyd/json_search.py

```diff
@@ -118,17 +167,539 @@ def __call__(self, response_dictionary):
+    acceptable_values = (
+        'high school', 'two-year college', 'college', 'university', 'college/university',
+        'post-grad', 'masters program', 'law school', 'med school', 'ph.d program'
+        )
+    types = "string or list of strings"
+    # TODO: also permit shorter key `education`?
+    @util.makelist_decorator
+    def transform(education_level=[]):
+        education = []
+        for k in education_level:
+            k = k.lower()
+            i = magicnumbers.maps.education_level[k]
+            if i == 1:
+                education.add('high_school'):
+            elif i == 2:
+                education.add('two_year_college'):
+            elif i == 3 or i == 4 or k == 'college/university':
+                education.add('college_university'):
+            elif i in (5,6,7,8) or k == 'post-grad' or k == 'post grad' or k == 'grad':
+                education.add('post_grad')
+            else:
+                # i == 9 when k == 'space camp', or unrecognized k
+                raise TypeError
+        return education
+
+
+class ChildrenFilter(search_filters.filter_class):
+    output_key = "children"
+    decide = search_filters.any_not_none_decider
+    acceptable_values = ((True, 'has a kid', 'has kids', False, "doesn't have a kid", "doesn't have kids"),
+                         (True, 'wants', 'wants kids', False, "doesn't want", "doesn't want kids",
+                         'might want', 'might want kids'))
+    # types = (str, str)
+    # TODO: also permit single key `kids`?
+    def transform(has_kids=None, wants_kids=None):
+        children = []
+        if isinstance(has_kids, six.string_types):
+            has_kids = has_kids.lower()
+        if isinstance(wants_kids, six.string_types):
+            wants_kids = wants_kids.lower()
+        if has_kids == 'has a kid' or has_kids == 'has kids' or has_kids is True:
+            children.append('has_one_or_more')
+        elif has_kids == "doesn't have a kid" or has_kids == "doesn't have kids" or has_kids is False:
+            children.append('doesnt_have')
+        elif has_kids is not None:
+            raise TypeError
+        # else: children.extend(('has_one_or_more', 'doesnt_have'))
+        if wants_kids == 'wants' or wants_kids == 'wants kids':
+            children.append('wants_kids')
+        elif wants_kids == "doesn't want" or wants_kids == "doesn't want kids":
+            children.append('doesnt_want')
+        elif wants_kids == 'might want' or wants_kids == 'might want kids':
+            children.append('might_want')
+        elif wants_kids is True:
+            children.extend(('wants_kids', 'might_want'))
+        elif wants_kids is False:
+            children.extend(('doesnt_want', 'might_want'))
+        elif wants_kids is not None:
+            raise TypeError
+        # else: children.extend(('wants_kids', 'might_want', 'doesnt_want'))
+        return children
+
+
+def normalform(keys, magicmap):
```

Comment:
this is a strange name for this. Why not just call it remap_items or something like that. Also use camel_case names for function definitions